File: nodejs/commons/src/dummy-task.ts
import type { Task } from './task';

export class DummyTask implements Task<string> {
  public message = '';

  async execute(message: string): Promise<void> {
    console.log('>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>');
    console.log('>>', new Date().toISOString());
    console.log('>>', message);
    console.log('>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>');
    this.message = message;
  }
}

File: java/commons/src/test/java/com/day1co/cornerstone/commons/DummyTask.java
package com.day1co.cornerstone.commons;

/**
 * 테스트용 타스크.
 *
 * 태스크를 실행하면 전달한 메시지가 속성에 설정된다.
 */
public class DummyTask implements Task<String> {
    public String message;

    @Override
    public Class<String> getMessageClass() {
        return String.class;
    }

    @Override
    public void execute(String message) {
        this.message = message;
    }
}
